angr/analyses/decompiler/structured_codegen/c.py

```diff
@@ -1892,63 +1915,116 @@ def default_simtype_from_size(self, n: int) -> SimType:
+                    else:
+                        l.warning("Unsupported variable offsetting operator %s. Default elem_size to 1.", offset.op)
+                        elem_size = 1
 
-                if type_size == elem_size:
-                    # awesome - no conversion needed
-                    offset = offset.lhs
+                    if type_size == elem_size:
+                        # awesome - no conversion needed
+                        idx = offset.lhs
+                    else:
+                        # damn - we need conversion
+                        if elem_size > type_size:
+                            coeff = CConstant(elem_size // type_size,
+                                              self.default_simtype_from_size(self.project.arch.bytes),
+                                              codegen=self)
+                            idx = CBinaryOp("Mul", offset.lhs, coeff, None, codegen=self)
+                        else:  # elem_size < type_size
+                            coeff = CConstant(type_size // elem_size,
+                                              self.default_simtype_from_size(self.project.arch.bytes),
+                                              codegen=self)
+                            idx = CBinaryOp("Div",
+                                               offset.lhs,
+                                               coeff,
+                                               None,
+                                               codegen=self)
                 else:
-                    # damn - we need conversion
-                    if elem_size > type_size:
-                        coeff = CConstant(elem_size // type_size,
-                                          self.default_simtype_from_size(self.project.arch.bytes),
-                                          codegen=self)
-                        offset = CBinaryOp("Mul",
-                                           offset.lhs,
-                                           coeff,
-                                           None,
-                                           codegen=self)
-                    else:  # elem_size < type_size
-                        coeff = CConstant(type_size // elem_size,
-                                          self.default_simtype_from_size(self.project.arch.bytes),
-                                          codegen=self)
-                        offset = CBinaryOp("Div",
-                                           offset.lhs,
-                                           coeff,
-                                           None,
-                                           codegen=self)
-            else:
-                # FIXME: unsupported
-                pass
+                    idx = None
+
+                if idx is not None:
+                    cvariable = self._cvariable(variable, variable_type=variable_type, tags=tags)
+                    return CIndexedVariable(cvariable, idx, tags=tags, codegen=self)
```

Comment:
These lines are just completely wrong. The first line triggers an infinite recursion loop and for the second one, CIndexedVariable doesn't take tags. I tried hacking around these issues but then I ran into issues where the code calling this function doesn't expect to take a CIndexedVariable, and it can't duck-type as a CVariable, so it crashes.

Reply:
Which test case triggers it?

Reply:
test_decompiling_amp_challenge03_arm